index.d.ts

```diff
@@ -0,0 +1,16 @@
+export type Scalar = number | string | boolean | bigint | null | undefined;
+export interface Err<Code extends Scalar, Ext extends Object> extends Error, Ext {
+  code: Code
+}
+
+interface CreateError {
+  <Code extends Scalar, Ext extends Object> (
+    reason:string|Error,
+    code:Code,
+    ext:Ext
+  ):Err<Code, Ext>
+  <Code extends Scalar> (reason:string|Error, code:Code):Err<Code, {}>
+}
```

Comment:
I think this is missing the `errCode(err, ext)` signature, see [Usage](https://github.com/IndigoUnited/js-err-code#usage) in the readme.